test/runtests.jl

```diff
@@ -321,3 +321,21 @@ end
     @test isapprox(lm(Xf, y).pp.beta0, ones(2))
     @test isapprox(lm(X, yf).pp.beta0, ones(2))
 end
+
+@testset "F test for model comparison" begin
+    d = DataFrame(Treatment=[1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 2.], Result=[1.1, 1.2, 1, 2.2, 1.9, 2, .9, 1, 1, 2.2, 2, 2], Other=[1, 1, 2, 1, 2, 1, 3, 1, 1, 2, 2, 1])
+    mod = lm(Result~Treatment, d).model
```

Comment:
All formulas should be in a `@formula()` call. The use of `~` on its own like this was removed in Julia 0.6.